test/e2e/suspend_test.go

```diff
@@ -0,0 +1,128 @@
+kind: Workflow
+metadata:
+  generateName: suspend-node-timeout-without-default-value-
+spec:
+  entrypoint: suspend
+  templates:
+  - name: suspend
+    steps:
+    - - name: approve
+        template: approve
+    - - name: release
+        template: whalesay
+        arguments:
+          parameters:
+            - name: message
+              value: "{{steps.approve.outputs.parameters.message}}"
+  - name: approve
+    suspend:
+      duration: 5s
+    outputs:
+      parameters:
+        - name: message
+          valueFrom:
+            supplied: {}
+  - name: whalesay
+    inputs:
+      parameters:
+        - name: message
+    container:
+      image: docker/whalesay
+      command: [cowsay]
+      args: ["{{inputs.parameters.message}}"]
+`).
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(fixtures.ToBeFailed).
+		Then().
+		ExpectWorkflow(func(t *testing.T, metadata *metav1.ObjectMeta, status *wfv1.WorkflowStatus) {
+			assert.Equal(t, wfv1.WorkflowFailed, status.Phase)
+			assert.Contains(t, "raw output parameter 'message' has not been set and does not have a default value", status.Message)
+		})
+}
+
+func (s *TestSuspendSitue) TestSuspendNodeTimeoutWithDefaultValue() {
+	s.Given().Workflow(`
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: suspend-node-timeout-with-default-value-
+spec:
+  entrypoint: suspend
+  templates:
+  - name: suspend
+    steps:
+    - - name: approve
+        template: approve
+    - - name: release
+        template: whalesay
+        arguments:
+          parameters:
+            - name: message
+              value: "{{steps.approve.outputs.parameters.message}}"
+  - name: approve
+    suspend:
+      duration: 5s
+    outputs:
+      parameters:
+        - name: message
+          valueFrom:
+            default: default message
+            supplied: {}
+  - name: whalesay
+    inputs:
+      parameters:
+        - name: message
+    container:
+      image: docker/whalesay
+      command: [cowsay]
+      args: ["{{inputs.parameters.message}}"]
+`).
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(fixtures.ToBeSucceeded).
+		Then().
+		ExpectWorkflow(func(t *testing.T, metadata *metav1.ObjectMeta, status *wfv1.WorkflowStatus) {
+			assert.Equal(t, wfv1.WorkflowSucceeded, status.Phase)
+			assert.Equal(t, status.Progress, wfv1.Progress("2/2"))
+		}).
+		ExpectWorkflowNode(func(status wfv1.NodeStatus) bool {
+			return status.Name == "suspend-node-timeout-with-default-value-955vn[0].approve"
+		}, func(t *testing.T, status *wfv1.NodeStatus, pod *apiv1.Pod) {
+			assert.Equal(t, wfv1.NodeSucceeded, status.Phase)
+			assert.Equal(t, 1, len(status.Outputs.Parameters))
+			assert.Equal(t, "message", status.Outputs.Parameters[0].Name)
+			assert.Equal(t, wfv1.AnyStringPtr("default message"), status.Outputs.Parameters[0].Value)
+		}).
+		ExpectWorkflowNode(func(status wfv1.NodeStatus) bool {
+			return status.Name == "suspend-node-timeout-with-default-value-955vn[1].release"
```

Comment:
is the `-955vn` deterministic, maybe should use `name` instead of `generateName`